### DISCUSSION TITLE
What causes SourceChanged event args' IsNewDocument to be false?

### DISCUSSION THREAD

**Discussion by benark** [MicrosoftEdge/WebView2Feedback#2431] [Q&A]
I am trying to track down an issue where navigating around a site (not public, unfortunately), will not trigger ContentLoading, NavigationCompleted, or NavigationStarting events after the initial page load. Subsequent loads of the same page will only trigger SourceChanged events and when the user is navigating around the site, the SourceChanged event args' IsNewDocument property is false. If the IsNewDocument property is true (which is more common on most public sites I've tested against), then the expected events are fired. Since the navigation-related events aren't firing, it is impossible to reliably interact with the page after it has loaded since SourceChanged is too early in the normal navigation sequence.

My question, which may ultimately result in creating an Issue, is what triggers WebView2 to set this IsNewDocument property in the SourceChanged event's args to false and is there any way to change that behavior to force it to properly fire the navigation events for every actual page navigation/change?

**Comment by snechaev** [comment]
I saw `IsNewDocument==false` in case of anchor navigation inside the current page like this `<a href="#my_anchor">text</a>`

**Comment by victorthoang** [comment]
hello @benark 

Are you still looking for an answer to your question? If you still need a response, we can turn this into an Issue with a Question tag (which is typically how we address questions and bugs.

The discussion thread might be more useful for off topic questions or non-issue related inquiries!

Let me know if you'd like me to create an issue from this discussion, or if you'd prefer to add it. Thanks!
